### ISSUE TITLE
Set up testing environment for server

### ISSUE THREAD

**Issue by karoliina** [karoliina/COMP4770#9]
We need a setup for easily writing and running unit tests for the server (the controllers, mostly). I'm going to use [this tutorial](https://semaphoreci.com/community/tutorials/a-tdd-approach-to-building-a-todo-api-using-node-js-and-mongodb) to set up the required libraries, write a few tests for the existing routes, and to create a test runner script. The tests will be using Mocha, Chai and Sinon.js.

**Comment by karoliina** [comment]
Okay, that tutorial actually sucks. [This one](https://codeutopia.net/blog/2016/06/10/mongoose-models-and-unit-tests-the-definitive-guide/) seems a little better (and funnier).

**Comment by karoliina** [comment]
I've made progress with this, using [supertest](https://github.com/visionmedia/supertest) (which uses SuperAgent internally, [useful docs](http://visionmedia.github.io/superagent/)). It straight up queries the server routes and the only difference to the "real" server instance is that I've set up a separate testing database that will need to be populated with appropriate before each test suite, and emptied afterwards. So these will essentially be integration tests instead of unit tests.

Some code is up on the requests-server-side branch. I'll write tests for all of the existing server routes, except maybe authentication for now, and will then create a PR. That PR will contain the database models for requests, groups and notifications as well.
